lib/travis/build/script/r.rb

```diff
@@ -0,0 +1,368 @@
+          setup_devtools
+          sh.echo "Installing R packages from github: #{packages.join(', ')}"
+          pkg_arg = packages_as_arg(packages)
+          install_script = [
+            'library("devtools")',
+            "options(repos = c(CRAN = \"#{config[:cran]}\"))",
+            "install_github(#{pkg_arg}, build_vignettes = FALSE)",
+          ].join('; ')
+          sh.cmd "Rscript -e '#{install_script}'"
+        end
+        
+        def r_binary_install(packages)
+          case config[:os]
+          when 'linux'
+            sh.echo "Installing *binary* R packages: #{packages.join(', ')}"
+            apt_install packages.collect{|p| "r-cran-#{p.downcase}"}
+          else
+            sh.echo "R binary packages not supported on #{config[:os]}, " +
+                    'falling back to source install'
+            r_install packages
+          end
+        end
+
+        def apt_install(packages)
+          pkg_arg = packages.join(' ')
+          sh.echo "Installing apt packages: #{packages.join(', ')}"
+          sh.cmd "sudo apt-get install #{pkg_arg}", retry: true
+        end
+
+        def bioc_install(packages)
+          if not needs_bioc?
+            return
+          end
+          setup_bioc
+          sh.echo "Installing bioc packages: #{packages.join(', ')}"
+          pkg_arg = packages_as_arg(packages)
+          install_script = [
+            "source(\"#{config[:bioc]}\")",
+            'options(repos=biocinstallRepos())',
+            "biocLite(#{pkg_arg})",
+          ].join('; ')
+          sh.cmd "Rscript -e '#{install_script}'"
+        end
+
+        def install_deps
+          setup_devtools
+          if not needs_bioc?
+            install_script = [
+              'library("devtools")',
+              "options(repos = c(CRAN = \"#{config[:cran]}\"))",
+              'install_deps(dependencies = TRUE)',
+            ].join('; ')
+          else
+            install_script = [
+              'library("devtools")',
+              'options(repos = biocinstallRepos())',
+              'install_deps(dependencies = TRUE)',
+            ].join('; ')
+          end
+          sh.cmd "Rscript -e '#{install_script}'"
+        end
+
+        def export_rcheck_dir
+          pkg_script = [
+            'library("devtools")',
+            'cat(paste0(as.package(".")$package, ".Rcheck"))',
+          ].join('; ')
+          sh.export 'RCHECK_DIR', "$(Rscript -e '#{pkg_script}')"
+        end
+        
+        def dump_logs
+          export_rcheck_dir
+          ['out', 'log', 'fail'].each do |ext|
+            cmd = [
+              'for name in',
+              "$(find \"${RCHECK_DIR}\" -type f -name \"*#{ext}\");",
+              'do',
+              'echo ">>> Filename: ${name} <<<";',
+              'cat ${name};',
+              'done',
+            ].join(' ')
+            sh.cmd cmd
+          end
+        end
+        
+        def setup_bioc
+          if not @bioc_installed
```

Comment:
nice! i didn't even know about `unless`.

i assume the convention is to keep with `if` in the case you actually have two branches? (i'm basing this on emacs lisp.)